### ISSUE TITLE
[BUG]: Unique type fields are not removed once a Model Item is removed

### ISSUE THREAD

**Issue by hasnainvirk** [sensedeep/dynamodb-onetable#83]
**Describe the bug**
Unique type fields are not removed once a Model Item having unique attrbiute is removed. 

**To Reproduce**
```
    models: {

        Tenant: {
            pk: { type: String, value: 'tenant#${id}' },
            sk: { type: String, value: 'tenant#' },
            id: { type: String, uuid: true, validate: Match.ulid },
            name: { type: String, required: true, unique: true, validate: Match.name },
            balance: { type: Number, default: 0 },

            //  Search by tenant name or by type
            gs1pk: { type: String, value: 'tenant#' },
            gs1sk: { type: String, value: 'tenant#${name}#${id}' },
        },
}
```
The attribute 'name' is marked as unique.

```
// API to remove
 await this.Tenant.remove({ id: id });
```

**Expected behavior**
Unique type fields should be removed alongwith the Model Item.

**Comment by mobsense** [comment]
Here is a quick patch if you need:

```
@@ -921,8 +921,8 @@ export class Model {
                     params.fallback = true
                     return properties
                 }
-                if (KeysOnly[op] && attribute != index.hash && attribute != index.sort) {
-                    //  Keys only for get and delete
+                if (KeysOnly[op] && attribute != index.hash && attribute != index.sort && !field.unique) {
+                    //  Keys only for get and delete. Must include unique properties.
                     //  FUTURE: could have a "strict" mode where we warn for other properties instead of ignoring.
                     continue
```

**Comment by mobsense** [comment]
NOTE: when you do the remove, you cannot just pass the ID only. You must pass the unique properties values too.

If the unique property uses value templates, you need to pass all properties used by the value template.

**Comment by mobsense** [comment]
Fixed in 1.6.2
